Core.ApplicationServices/UserService.cs

```diff
@@ -306,31 +306,58 @@ public void Dispose()
                 .Transform(Result<IQueryable<User>, OperationError>.Success);
         }
 
-        public Maybe<OperationError> DeleteUserFromKitos(Guid userUuid)
+        public Maybe<OperationError> InitiateUserDeletion(Guid userUuid, int? scopedToOrganizationId = null)
         {
-            using var transaction = _transactionManager.Begin();
-
             var user = _userRepository.AsQueryable().ByUuid(userUuid);
             if (user == null)
                 return new OperationError(OperationFailure.NotFound);
             if(_organizationalUserContext.UserId == user.Id)
                 return new OperationError("You cannot delete a user you are currently logged in as", OperationFailure.Forbidden);
 
+            if (scopedToOrganizationId.HasValue && scopedToOrganizationId.Value != 0)
+                return DeleteUserFromOrganizationOrKitos(scopedToOrganizationId.Value, user);
+            
+            return _authorizationContext.AllowDelete(user) 
+                ? DeleteUserFromKitos(user) 
+                : new OperationError(OperationFailure.Forbidden);
+        }
 
-            if (!_authorizationContext.AllowDelete(user))
-                return new OperationError(OperationFailure.Forbidden);
+        private Maybe<OperationError> DeleteUserFromKitos(User user)
+        {
+            using var transaction = _transactionManager.Begin();
             
             _domainEvents.Raise(new EntityBeingDeletedEvent<User>(user));
 
             Delete(user);
             _userRepository.Save();
             transaction.Commit();
-            
+
             _domainEvents.Raise(new AdministrativeAccessRightsChanged(user.Id));
 
             return Maybe<OperationError>.None;
         }
 
+        private Maybe<OperationError> DeleteUserFromOrganizationOrKitos(int scopedToOrganizationId, User user)
+        {
+            var deletionStrategy = GetUserDeletionStrategy(scopedToOrganizationId, user);
+            if (deletionStrategy.Failed)
+                return deletionStrategy.Error;
+
+            switch (deletionStrategy.Value)
+            {
+                case UserDeletionStrategy.Global:
+                    var organization = _orgRepository.AsQueryable().ById(scopedToOrganizationId);
+                    return _authorizationContext.AllowModify(organization) 
```

Comment:
you dont have to reauthorize here because you already did that check when you asked for strategies right?
If not strategies, then you are not authorized to do anything